dbt/include/athena/macros/adapters/relation.sql

```diff
@@ -1,8 +1,32 @@
 {% macro athena__drop_relation(relation) -%}
+  {%- set native_drop = config.get('drop', default='False') | as_bool -%}
+  {%- set natively_droppable = config.get('table_type', default='hive') == 'iceberg' or relation.type == 'view' -%}
```

Comment:
This unfortunately doesn't work in all cases. The source of truth is not what we have in the config but what we have in the glue catalog. We had this implementation back then, and it leads to some issues.
Have a look at this refactor https://github.com/dbt-athena/dbt-athena/pull/249

Specifically, we called
`{% set rel_type = adapter.get_table_type(relation.schema, relation.table) %}` to fetch table type via an adapter method
(the method doesn't exist anymore), we need to put it back (cc @Jrmyy) and use the [this](https://github.com/dbt-athena/dbt-athena/blob/main/dbt/adapters/athena/relation.py#L92-L104) helper to get the right enum type, then we can check if we can drop natively or not.

There are edge cases where this is needed, for example if have a relation that is an hive table, and then the config is changed to iceberg table, we want the adapter drop the table and then clean up s3, but based in this implementation config is used, there the s3 clean will be used, leading to some dirty s3 path - same with views.
Long story short, the issue is that we need to rely on what "we" have (data catalog), not what we plan to have after run (config).

Reply:
Applied changes:
- picked up old impl method
- picked up old tests
- replaced config get with table type get
- confirmed as best I can in logs, tests all pass